pkg/archive/paths.go

```diff
@@ -15,24 +15,19 @@
 package archive
 
 import (
+	"archive/tar"
 	"archive/zip"
 	"context"
 	"io"
-	"os"
 )
 
-// A WalkFn iterates through an archive, using TarReaderProvider to read the archive type, calling FileWalkFn on each member file.
-type WalkFn func(ctx context.Context, path string, getTarReader TarReaderProvider, walkFn FileWalkFn) error
-
-// A ZipWalkFn iterates through an zip, calling FileWalkFn on each member file.
-type ZipWalkFn func(ctx context.Context, r *zip.Reader, walkFn FileWalkFn) error
-
-// ZipReadCloserProvider should open a zip.ReadCloser when provided with the given path.
-type ZipReadCloserProvider func(path string) (*zip.ReadCloser, error)
+// A WalkFn iterates through an archive, calling FileWalkFn on each member file.
+type WalkFn func(ctx context.Context, walkFn FileWalkFn) error
 
 // FileWalkFn is called by a WalkFn on each file contained in an archive.
 type FileWalkFn func(ctx context.Context, path string, size int64, contents io.Reader) (proceed bool, err error)
 
+// WalkZipFiles walks the files in the provided *zip.Reader, calling walkFn on each header.
```

Comment:
Function can now be private (or even inlined into the one calling location) -- previously this was exposed since it was referenced directly, but no more need to do that with this refactor.